### PR TITLE
support fixed constraint

### PR DESCRIPTION
**Pull request by yiwenxue**
Re: #https://github.com/cocos/3d-tasks/issues/12930#issue-1290969991

### Changelog

* support fixed constraint

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/physics/cannon/cannon-shared-body.ts [hunk 1]
```diff
@@ -92,10 +96,10 @@ export class CannonSharedBody {
...
 
     /**
-     * add or remove from world \
-     * add, if enable \
-     * remove, if disable & shapes.length == 0 & wrappedBody disable
-     */
+      * add or remove from world \
+      * add, if enable \
+      * remove, if disable & shapes.length == 0 & wrappedBody disable
+      */
```
Comment (lealzhan) [inline cocos/physics/cannon/cannon-shared-body.ts:102]: seems useless modification

File: cocos/physics/cannon/cannon-shared-body.ts [hunk 2]
```diff
@@ -105,7 +109,7 @@ export class CannonSharedBody {
             }
         } else if (this.index >= 0) {
             const isRemove = (this.wrappedShapes.length === 0 && this.wrappedBody == null)
-                || (this.wrappedShapes.length === 0 && this.wrappedBody != null && !this.wrappedBody.isEnabled);
+                 || (this.wrappedShapes.length === 0 && this.wrappedBody != null && !this.wrappedBody.isEnabled);
```
Comment (lealzhan) [inline cocos/physics/cannon/cannon-shared-body.ts:112]: seems useless modification

File: cocos/physics/bullet/constraints/bullet-fixed-constraint.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    setBreakTorque (v: number): void {
+        // not supported
+    }
+
+    get constraint (): FixedConstraint {
+        return this._com as FixedConstraint;
+    }
+
+    onComponentSet () {
```
Comment (lealzhan) [inline cocos/physics/bullet/constraints/bullet-fixed-constraint.ts:50]: this implementation seems a little different fro  another two constraint implementations

File: cocos/physics/bullet/constraints/bullet-fixed-constraint.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        const cb = this.constraint.connectedBody;
+        const bodyA = (this._rigidBody.body as BulletRigidBody).sharedBody;
+        const bodyB = cb ? (cb.body as BulletRigidBody).sharedBody : (PhysicsSystem.instance.physicsWorld as BulletWorld).getSharedBody(bodyA.node);
+
+        const pos : Vec3 = CC_V3_0;
+        const rot : Quat = CC_QUAT_0;
+        const trans0 = BulletCache.instance.BT_TRANSFORM_0;
+        const trans1 = BulletCache.instance.BT_TRANSFORM_1;
+        const quat = BulletCache.instance.BT_QUAT_0;
+
```
Comment (lealzhan) [inline cocos/physics/bullet/constraints/bullet-fixed-constraint.ts:72]: add some comments ?

File: cocos/physics/cannon/cannon-shared-body.ts [hunk 5]
```diff
@@ -50,9 +50,9 @@ const CollisionEventObject = {
 };
 
 /**
- * node : shared-body = 1 : 1
- * static
- */
+  * node : shared-body = 1 : 1
```
Comment (lealzhan) [inline cocos/physics/cannon/cannon-shared-body.ts:53]: seems useless modification

File: native/cocos/physics/physx/joints/PhysXFixedJoint.cpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+void PhysXFixedJoint::onComponentSet() {
+    _transA = physx::PxTransform{physx::PxIdentity};
+    _transB = physx::PxTransform{physx::PxIdentity};
+
+    physx::PxRigidActor *actor0 = _mSharedBody->getImpl().rigidActor;
+    physx::PxRigidActor *actor1 = nullptr;
+
+    if (_mConnectedBody) {
+        auto *actor1 = _mConnectedBody->getImpl().rigidActor;
```
Comment (lealzhan) [inline native/cocos/physics/physx/joints/PhysXFixedJoint.cpp]: ?

File: native/cocos/physics/physx/joints/PhysXFixedJoint.cpp [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    pos = _mSharedBody->getNode()->getWorldPosition();
+    rot = _mSharedBody->getNode()->getWorldRotation();
+    pxSetVec3Ext(_transA.p, pos);
+    pxSetQuatExt(_transA.q, rot);
+
+    physx::PxRigidActor *actor0 = _mSharedBody->getImpl().rigidActor;
+    physx::PxRigidActor *actor1 = nullptr;
+
+    if (_mConnectedBody) {
+        auto *actor1 = _mConnectedBody->getImpl().rigidActor;
```
Comment (lealzhan) [inline native/cocos/physics/physx/joints/PhysXFixedJoint.cpp]: ?

Comment (lealzhan) [reply native/cocos/physics/physx/joints/PhysXFixedJoint.cpp]: connectBody's rigidActor might be null?

File: cocos/physics/cannon/cannon-shared-body.ts (hunk 1 above)
Comment (yiwenxue) [reply cocos/physics/cannon/cannon-shared-body.ts:102]: ESlint formatting

File: cocos/physics/cannon/cannon-shared-body.ts (hunk 2 above)
Comment (yiwenxue) [reply cocos/physics/cannon/cannon-shared-body.ts:112]: ESlint formatting

File: cocos/physics/cannon/cannon-shared-body.ts (hunk 5 above)
Comment (yiwenxue) [reply cocos/physics/cannon/cannon-shared-body.ts:53]: ESlint formatting

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

File: native/cocos/physics/physx/joints/PhysXFixedJoint.cpp (hunk 7 above)
Comment (yiwenxue) [reply native/cocos/physics/physx/joints/PhysXFixedJoint.cpp]: It's possible, but not expected. If the connected body is null, the constraint will be connected to a fixed object (the world frame).

File: native/cocos/physics/physx/joints/PhysXFixedJoint.cpp (hunk 6 above)
Comment (yiwenxue) [reply native/cocos/physics/physx/joints/PhysXFixedJoint.cpp]: fixed

**Review by lealzhan — APPROVED** [review]

**Pull request merged**